### PR TITLE
Filter binary symbols to libc/libSystem origin

### PR DESCRIPTION
**Pull request by isseis**
## Summary

- Add `CategorySyscallWrapper` for non-network libc/libSystem symbols, so `DetectedSymbols` now carries all libc/libSystem imports (not just network ones)
- ELF analyzer: use VERNEED per-symbol library attribution; fall back to DT_NEEDED only when libc is the sole imported library
- Mach-O analyzer: use two-level namespace ordinals to attribute symbols to libSystem; handle flat-namespace binaries (Go CGO) and Symtab-absent fallback
- `network_analyzer.go`: switch network detection from `len(DetectedSymbols) > 0` to `IsNetworkCategory(sym.Category)` so `syscall_wrapper` symbols do not trigger `NetworkDetected`
- Fix DRY violation in `analyzeSliceFallback` (reimplemented `hasLibSystem` loop → call the function); precompute `libSystemPresent` before symbol loop

## Test plan

- [ ] `make test` passes (all packages)
- [ ] `make lint` passes (0 issues)
- [ ] `TestIsNetworkCategory` — new unit test for `IsNetworkCategory`
- [ ] `TestBuildDetectedSymbols` — covers VERNEED path, fallback path, dlopen independence
- [ ] `TestHasOnlyLibcImportedLibraries` / `TestBuildDetectedSymbols_Fallback*` — ELF fallback edge cases
- [ ] `TestIsLibSystemLibrary` / `TestIsFlatNamespace` — Mach-O helper boundary tests
- [ ] `TestAnalyzeSlice_NonLibSystemSymbolsExcluded` — verifies `/usr/bin/curl` does not trigger `NetworkDetected` at symbol level
- [ ] `TestIsNetworkViaBinaryAnalysis_SyscallWrapperOnly` — verifies `syscall_wrapper`-only symbols do not trigger `NetworkDetected`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/runner/security/machoanalyzer/standard_analyzer.go [hunk 1]
```diff
@@ -52,48 +54,206 @@ func isMachOMagic(b []byte) bool {
...
+	}
+	if f.Dysymtab != nil {
+		dt := f.Dysymtab
+		total := uint32(len(f.Symtab.Syms)) //#nosec G115 -- len() is non-negative and slice lengths fit in uint32
+		if dt.Iundefsym > total {
+			return nil
+		}
+		end := dt.Iundefsym + dt.Nundefsym
+		if end > total {
+			end = total
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/machoanalyzer/standard_analyzer.go:155]: Fixed: added overflow check as suggested.

File: internal/runner/security/elfanalyzer/standard_analyzer.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 	"io"
 	"log/slog"
 	"os"
+	"strings"
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/elfanalyzer/standard_analyzer.go:12]: Fixed: added path/filepath to imports and updated isLibcLibrary to use filepath.Base.

File: internal/runner/security/elfanalyzer/standard_analyzer.go [hunk 3]
```diff
@@ -246,45 +234,169 @@ func (a *StandardELFAnalyzer) AnalyzeNetworkSymbols(path string, contentHash str
...
+		return binaryanalyzer.AnalysisOutput{Result: binaryanalyzer.NoNetworkSymbols}
+	}
+
+	// For binaries without VERNEED, allow the DT_NEEDED fallback only when
+	// libc is the sole imported library. If other libraries are present, we
+	// cannot safely attribute every STT_FUNC import to libc.
+	fallbackAllFuncsFromLibc := false
+	if !hasVERNEED {
+		libs, _ := elfFile.ImportedLibraries()
+		fallbackAllFuncsFromLibc = hasOnlyLibcImportedLibraries(libs)
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/elfanalyzer/standard_analyzer.go]: Fixed: ImportedLibraries() error is now handled; returns AnalysisError when it fails.

File: internal/runner/security/elfanalyzer/standard_analyzer.go [hunk 4]
```diff
@@ -246,45 +234,169 @@ func (a *StandardELFAnalyzer) AnalyzeNetworkSymbols(path string, contentHash str
...
+
+	return detected, dynamicLoadSyms
+}
+
+// isLibcLibrary returns true if the given library name matches a known libc pattern.
+// Recognized patterns: glibc ("libc.so.6") and musl ("libc.musl-<arch>.so.1").
+func isLibcLibrary(lib string) bool {
+	return strings.HasPrefix(lib, "libc.so.") ||
+		strings.HasPrefix(lib, "libc.musl-")
+}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/elfanalyzer/standard_analyzer.go:386]: Fixed: isLibcLibrary now uses filepath.Base to handle absolute paths in DT_NEEDED entries.

File: internal/runner/security/machoanalyzer/standard_analyzer.go [hunk 5]
```diff
@@ -52,48 +54,206 @@ func isMachOMagic(b []byte) bool {
 	return false
 }
 
-// analyzeSlice performs symbol and svc #0x80 analysis on a single *macho.File.
+// analyzeSlice performs libSystem-filtered symbol analysis on a single *macho.File.
 // Returns the AnalysisOutput for that slice.
 func (a *StandardMachOAnalyzer) analyzeSlice(f *macho.File) binaryanalyzer.AnalysisOutput {
+	// Retrieve referenced library list for ordinal resolution.
+	libs, _ := f.ImportedLibraries()
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/machoanalyzer/standard_analyzer.go]: Fixed: ImportedLibraries() error is now handled; returns AnalysisError when it fails.

[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/machoanalyzer/standard_analyzer.go]: Fixed: ImportedLibraries() error is now handled; returns AnalysisError when it fails.

File: internal/runner/security/elfanalyzer/standard_analyzer.go [hunk 6]
```diff
@@ -246,45 +234,169 @@ func (a *StandardELFAnalyzer) AnalyzeNetworkSymbols(path string, contentHash str
...
+	if !hasAnyUndef {
+		return binaryanalyzer.AnalysisOutput{Result: binaryanalyzer.NoNetworkSymbols}
+	}
+
+	// For binaries without VERNEED, allow the DT_NEEDED fallback only when
+	// libc is the sole imported library. If other libraries are present, we
+	// cannot safely attribute every STT_FUNC import to libc.
+	fallbackAllFuncsFromLibc := false
+	if !hasVERNEED {
+		libs, _ := elfFile.ImportedLibraries()
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/elfanalyzer/standard_analyzer.go]: Fixed: ImportedLibraries() error is now handled; returns AnalysisError when it fails.

File: internal/runner/security/elfanalyzer/analyzer_test.go [hunk 7]
```diff
@@ -141,102 +141,221 @@ func TestHasDynamicLoad_ELF(t *testing.T) {
...
+// collection in buildDetectedSymbols. AC-1, AC-2.
+func TestBuildDetectedSymbols(t *testing.T) {
+	networkSymbols := binaryanalyzer.GetNetworkSymbols()
 
 	tests := []struct {
 		name                    string
 		symbols                 []elf.Symbol
-		wantResult              binaryanalyzer.AnalysisResult
+		hasVERNEED              bool
+		libcInNeeded            bool
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/elfanalyzer/analyzer_test.go]: Fixed: renamed the struct field from libcInNeeded to fallbackAllFuncsFromLibc throughout the test.

File: docs/tasks/0106_symbol_library_filter/04_implementation_plan.md [hunk 8]
```diff
@@ -70,11 +70,11 @@
...
 
 **作業内容**:
-- [ ] library ordinal を使う libSystem 判定ヘルパーを追加する
-- [ ] `NormalizeSymbolName` を通したカテゴリ付与を統合する
-- [ ] Symtab なし時の `ImportedLibraries()` / `ImportedSymbols()` フォールバックを整理する
-- [ ] libSystem 以外のシンボルが記録されないことを確認するテストを追加する
-- [ ] AC-3 を満たすテストを追加または更新する
+- [x] library ordinal を使う libSystem 判定ヘルパーを追加する
+- [x] `NormalizeSymbolName` を通したカテゴリ付与を統合する
+- [x] Symtab なし時の `ImportedLibraries()` / `ImportedSymbols()` フォールバ ックを整理する
```
[the remark this answers is not in the document]
Comment (isseis) [reply docs/tasks/0106_symbol_library_filter/04_implementation_plan.md]: Fixed: removed the stray space from フォールバック.

File: internal/runner/security/machoanalyzer/standard_analyzer.go [hunk 9]
```diff
@@ -52,48 +54,206 @@ func isMachOMagic(b []byte) bool {
...
+		nUndf     = 0x0  // N_UNDF
+		nExt      = 0x01 // N_EXT (external)
+		nStab     = 0xe0 // stab mask
+	)
+	if f.Symtab == nil {
+		return nil
+	}
+	if f.Dysymtab != nil {
+		dt := f.Dysymtab
+		total := uint32(len(f.Symtab.Syms)) //#nosec G115 -- len() is non-negative and slice lengths fit in uint32
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/machoanalyzer/standard_analyzer.go]: Fixed: updated to // #nosec G115 (with a space after //).
